apps/dashboard/src/hooks/use-remove-grammarly.ts

```diff
@@ -0,0 +1,29 @@
+import { useRef, useCallback } from 'react';
+
+import { useMutationObserver } from './use-mutation-observer';
+
+export function useRemoveGrammarly<T extends HTMLElement>() {
+  const target = useRef<T>(null);
+  const handleGrammarlyRemoval = useCallback((mutations: MutationRecord[]) => {
+    for (const mutation of mutations) {
+      if (mutation.type === 'childList') {
+        for (const node of mutation.addedNodes) {
+          const isGrammarlyElement =
+            node instanceof HTMLElement && node.shadowRoot && node.tagName === 'GRAMMARLY-EXTENSION';
+
+          if (isGrammarlyElement) {
+            node.remove();
+          }
+        }
+      }
```

Comment:
@LetItRock do you see any performance downsides for this approach?

Reply:
I'm asking because I wouldn't want to negatively impact 80% of our users only because of 20% that do use Grammarly

Reply:
I don't expect the Maily DOM tree to be "huge" and "frequently" changing, which could potentially then cause performance problems. Also, we don't execute any code that blocks the main thread.

Reply:
Makes sense